src/video/v9990/V9990.cc

```diff
@@ -166,9 +172,10 @@ byte V9990::readIO(word port, EmuTime::param time)
 		case KANJI_ROM_1:
 		case KANJI_ROM_2:
 		case KANJI_ROM_3:
-		case REGISTER_SELECT:
-		case SYSTEM_CONTROL:
 		default:
+			if (writeOnly) {
+				getCliComm().printWarning("Reading from the V9990 write-only port ", portName[port]);
+			}
 			return peekIO(port, time);
 		}
 	}();
```

Comment:
I would suggest to write this as:
```
byte result = [&] {
    switch (port) {
    case COMMAND_DATA:
        return ...;
    case SYSTEM_CONTROL:
    case REGISTER_SELECT:
    case KANJI_ROM_0:
    case KANJI_ROM_2:
        getCliComm().printWarning("Reading from the V9990 write-only port ", portName[port]);
        [[fallthrough]];
    ...
    default:
        return peekIO(port, time);
}();
```
In other words: instead of setting and later testing a boolean, remove that boolean and directly print the message in the only place where that boolean was set.

I think also the ports KANJI_ROM_0 and KANJI_ROM_2 are write-only, but please verify.